src/remotecontrol.cpp

```diff
@@ -77,12 +79,13 @@ void RemoteControl::handle()
         effectManager.ClearRemoteColor();
         effectManager.SetInterval(0);
         effectManager.StartEffect();
-        g_Values.Brightness = 255;
+        g_ptrSystem->DeviceConfig().SetBrightness(BRIGHTNESS_MAX);
         return;
     }
     else if (IR_OFF == result)
     {
-        g_Values.Brightness = std::max(MIN_BRIGHTNESS, (int) g_Values.Brightness - BRIGHTNESS_STEP);
+        auto& deviceConfig = g_ptrSystem->DeviceConfig();
+        deviceConfig.SetBrightness((int)deviceConfig.GetBrightness() - BRIGHTNESS_STEP);

```

Comment:
This invocation uses the fact that DeviceConfig::SetBrightness accepts a plain int, and keeps the actual brightness within limits.

Reply:
I would suggest setting the reference variable to device config earlier in the code. I'd suggest before the if logic, such as line 41 and move "decode_results results;" down a line.

I would also suggest not setting to full power when the power button is pressed. The drawing system should just reference the last value set.

Reply:
Ok, duly noted. :)

> I would suggest setting the reference variable to device config earlier in the code. I'd suggest before the if logic, such as line 41 and move "decode_results results;" down a line.

Both of these concern personal preferences in coding (style). You're totally entitled to yours, I code things this way. In the case of deviceConfig, there is a specific reason I declare a variable (because I otherwise have to pull it from `*g_ptrSystem` twice), and I like my variables to have the smallest scope possible - in this case the `if (IR_OFF == result)` code block. the `decode_results` thing I didn't touch, i.e.: I left that where it is.

> I would also suggest not setting to full power when the power button is pressed. The drawing system should just reference the last value set.

That's how the remote control logic in the current codebase works. I'm not modifying that; I'm merely making the brightness a JSON-persisted device setting.